src/rgw/rgw_crypt.cc

```diff
@@ -1170,10 +1170,8 @@ int rgw_s3_prepare_encrypt(req_state* s, optional_yield y,
           return -EINVAL;
         }
         /* try to retrieve actual key */
-        std::string key_selector = create_random_key_selector(s->cct);
```

Comment:
Can you please provide a reference why this change is correct?

Reply:
There are no references to the keysel attr.

Reply:
Looks to be as `RGW_ATTR_CRYPT_KEYSEL` is used in ``reconstitute_actual_key_from_kms()`` that is called from below ``make_actual_key_from_kms()`` if KMS backend is the testing backend (``RGW_SSE_KMS_BACKEND_TESTING``)

Reply:
That is only referenced by the testing engine as you mentioned, not by SSE-S3 and kms that I'm deleting.

Reply:
IIUC so this breaks the testing backend, since in ``reconstitute_actual_key_from_kms()`` it will try to grab keysel from `attrs` but you've removed it here, so needs to move ``create_random_key_selector()`` all the way down in that function if you want to drop it here?

Reply:
Sorry, I couldn't follow. I'm removing the keysel from being set on objects where they are going to be encrypted by aws:kms and sse-s3 encryption modes. for testing engine, we still have this attr set, so `reconstitute_actual_key_from_kms()` would work properly.
`create_random_key_selector()` is being called in testing engine case and being set as attr: https://github.com/ceph/ceph/blob/6c0b03b9eadd30c9006af33257579484e58ff930/src/rgw/rgw_crypt.cc#L1279-L1280